### DISCUSSION TITLE
Any way to disable the file output?

### DISCUSSION THREAD

**Discussion by ovalofficer** [sherlock-project/sherlock#2247] [Q&A] [answered]
Hi, I'm making a web wrapper for Sherlock but I want to disable the file output. Is there a switch I can use to turn it off without editing Sherlock's source?

I couldn't find out how to do it. Sorry if it's something obvious

**Comment by ppfeister** [comment]
That's actually a good question. We don't have much user journey stuff documented yet.

Are you using Sherlock as a library, or are you interacting with it as if the web app were a normal command line user?

**Reply by ovalofficer** [reply]
To be specific, I'm using Flask for a simple API and using subprocess to call Sherlock with the supplied parameters. So, the latter. I'll make a frontend for it eventually. My goal is to be able to use Sherlock 'as-is' and not have to modify it at all.

**Reply by ppfeister** [reply]
If you're calling Sherlock as described, your best error-less bet would be to use a temp directory (perhaps via mktemp -d), and clearing it occasionally (or automatically). However, this wouldn't result in instant deletion just by itself, and would either be on reboot or when temp is automatically cleared.

You could also redirect to /dev/null via --output to prevent writes in the first place, but that will either throw a NotADirectory type of error or an IsADirectory type of error, since /dev/null isn't really _normal_. If you can catch this, that would be less-bad I suppose. This could work if that's acceptable.

Between the error handling and use of subprocess, you should really consider just using Sherlock as a library. No need to modify Sherlock itself -- it's available on PyPI as a proper dependency for importing [__(as here)__](https://github.com/ppfeister/sylva/blob/c6be20529cc75f103dd2fa218b1f331299a5551d/pyproject.toml#L9). I have an implementation of it over in Sylva that could make for a decent reference. The actual prep work and query is only about 10 lines long [__[here]__](https://github.com/ppfeister/sylva/blob/c6be20529cc75f103dd2fa218b1f331299a5551d/src/sylva/modules/sherlock.py#L33-L43). The remainder of the file is just handling the data it returned, so you can see a bit how that was done in my case.

**Reply by ovalofficer** [reply]
Appreciate the advice here. Seeing the implementation actually cleared it up a lot for me, the Sylva source is very readable. As for dealing with the created files, I'd considered both of those 😅. I'm sure they'd work, but they seem like work-arounds for what I feel should just be a switch in Sherlock. 

Early on in my project I had to find a way to deal with Sherlock's output being formatted as it is with the bullet points, and eventually found that grabbing the URLs with regex from the STDOUT was the best. A simple switch to just skip the formatting and output the found URLs would save a whole step. I could even see it being better for the user in some cases. 

Regardless, using Sherlock as a library will almost definitely make things easier. Again, appreciate the advice and the snippet link.

**Reply by ppfeister** [reply]
I don't believe there's an issue open for it, but I have been wanting to change things in that regard

My personal preference would be no file out by default, allowing for it via the already-existing flag, but that's undecided. In either case, if the _default_ is changed, that would likely be a 0.16.0 change since it could _theoretically_ break some existing workflows

Unless @sdushantha has any thoughts on adding a dedicated flag in the interim?

**Reply by sdushantha** [reply]
I'd be open to have no file output as the default fro 0.16.0. In the interim we can have a flag to disable the creation of the files

**Reply by ntexe** [reply, marked as answer]
#2263

**Reply by ppfeister** [reply]
Thanks @ntexe for the proposal!

@ovalofficer Are you pulling from PyPI or some other method of install? Might be able to push a patch if needed so it's available on PyPI and DH. If you're building from source, then it's now in master.

**Reply by ovalofficer** [reply]
Hi!
Yeah, I just use PyPI. Thanks so much y'all for doing this, I think it'll be useful for a lot of people! Seeing the implementation I could've contributed that myself 😅 I'm still a relatively new programmer so big codebases like this are spooky to me.
